web/src/repo/releases/RepositoryReleasesTagsPage.tsx

```diff
@@ -1,41 +1,57 @@
-import * as React from 'react'
-import { RouteComponentProps } from 'react-router-dom'
+import React, { useEffect, useMemo } from 'react'
+import * as H from 'history'
 import * as GQL from '../../../../shared/src/graphql/schema'
 import { FilteredConnection, FilteredConnectionQueryArgs } from '../../components/FilteredConnection'
 import { PageTitle } from '../../components/PageTitle'
 import { eventLogger } from '../../tracking/eventLogger'
-import { GitRefNode, queryGitRefs } from '../GitRef'
+import { GitRefNode, queryGitRefs as _queryGitRefs } from '../GitRef'
 import { RepositoryReleasesAreaPageProps } from './RepositoryReleasesArea'
 import { Observable } from 'rxjs'
 
-interface Props extends RepositoryReleasesAreaPageProps, RouteComponentProps<{}> {}
+interface Props extends RepositoryReleasesAreaPageProps {
+    history: H.History
+    location: H.Location
+    queryGitRefs?: (args: {
+        repo: GQL.ID
+        first?: number
+        query?: string
+        type: GQL.GitRefType
+        withBehindAhead?: boolean
+    }) => Observable<GQL.IGitRefConnection>
+}
 
 /** A page that shows all of a repository's tags. */
-export class RepositoryReleasesTagsPage extends React.PureComponent<Props> {
-    public componentDidMount(): void {
+export const RepositoryReleasesTagsPage: React.FunctionComponent<Props> = ({
+    repo,
+    history,
+    location,
+    queryGitRefs = _queryGitRefs,
+}) => {
+    useEffect(() => {
         eventLogger.logViewEvent('RepositoryReleasesTags')
-    }
+    }, [])
 
-    public render(): JSX.Element | null {
-        return (
-            <div className="repository-releases-page">
-                <PageTitle title="Tags" />
-                <FilteredConnection<GQL.IGitRef>
-                    className=""
-                    listClassName="list-group list-group-flush"
-                    noun="tag"
-                    pluralNoun="tags"
-                    queryConnection={this.queryTags}
-                    nodeComponent={GitRefNode}
-                    defaultFirst={20}
-                    autoFocus={true}
-                    history={this.props.history}
-                    location={this.props.location}
-                />
-            </div>
-        )
-    }
+    const queryTags = useMemo(
+        () => (args: FilteredConnectionQueryArgs): Observable<GQL.IGitRefConnection> =>
+            queryGitRefs({ ...args, repo: repo.id, type: GQL.GitRefType.GIT_TAG }),
+        [repo.id, queryGitRefs]
+    )
```

Comment:
Isn't this the same?
```suggestion
    const queryTags = useCallback(
        (args: FilteredConnectionQueryArgs): Observable<GQL.IGitRefConnection> =>
            queryGitRefs({ ...args, repo: repo.id, type: GQL.GitRefType.GIT_TAG }),
        [repo.id, queryGitRefs]
    )
```

Reply:
😬 true